run_project_tests.py

```diff
@@ -380,7 +380,12 @@ def run_tests(extra_args):
     build_time = 0
     test_time = 0
 
-    executor = conc.ProcessPoolExecutor(max_workers=multiprocessing.cpu_count())
+    try:
+        # This fails in some CI environments for unknown reasons.
+        num_workers = multiprocessing.cpu_count()
+    except Exception:
```

Comment:
Because `Exception` catches all exceptions including syntax errors, the module not having that function anymore, the module not being imported anymore by mistake, the interpreter crashing in a recoverable way, and so on. It masks bugs.

It's almost always better to be explicit about what you are catching especially in this case where the error only happens on CIs and because the error happens while running tests which is not a user-facing part of Meson.